### PR TITLE
wpaperd: 0.3.0 -> 1.0.1 & mv to by-name

### PR DESCRIPTION
**Pull request by fsnkty**
## Description of changes
update src to 1.0.1
use `cargoHash` over including the lock
include egl and wayland-client deps for build
add nu-nu-ko to maintainers

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux

~~- [ ] Tested, as applicable:~~ no tests.

- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
 
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nu-nu-ko** [comment]
oh oops I always forget something like that when moving to by-name

**Review by DPDmancul — APPROVED** [review]

**Review by DPDmancul — APPROVED** [review]

File: pkgs/by-name/wp/wpaperd/package.nix [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
   pname = "wpaperd";
-  version = "0.3.0";
+  version = "1.0.1";
 
   src = fetchFromGitHub {
     owner = "danyspin97";
     repo = pname;
     rev = version;
-    sha256 = "cgjHCSBrkX3aoz42qBS/1JUGhc7sZKarKByntp7ubaQ=";
+    sha256 = "sha256-5riZ/6yjgsW++SUIyJP5rFG65tkjJKgtvDLIGaoiHN0=";
```
Comment (Aleksanaa) [inline pkgs/by-name/wp/wpaperd/package.nix:11]:
```suggestion
    hash = "sha256-5riZ/6yjgsW++SUIyJP5rFG65tkjJKgtvDLIGaoiHN0=";
```

File: pkgs/by-name/wp/wpaperd/package.nix [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
+{ lib, rustPlatform, fetchFromGitHub, pkg-config, libxkbcommon, wayland, libGL }:
 
 rustPlatform.buildRustPackage rec {
   pname = "wpaperd";
-  version = "0.3.0";
+  version = "1.0.1";
 
   src = fetchFromGitHub {
     owner = "danyspin97";
     repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/wp/wpaperd/package.nix:9]:
```suggestion
    repo = "wpaperd";
```

Comment (nu-nu-ko) [reply pkgs/by-name/wp/wpaperd/package.nix:9]: whats the reasoning for this change? pname being the repo name seems reasonable

Comment (Aleksanaa) [reply pkgs/by-name/wp/wpaperd/package.nix:9]: See https://github.com/NixOS/nixpkgs/issues/277994

**Comment by nu-nu-ko** [comment]
@nbraud Hopefully that last force push works. I'd just like to point out another PR from r-ryantm that currently depends on this change but is otherwise ready to merge (https://github.com/NixOS/nixpkgs/pull/305629),
thank you.

**Review by DPDmancul — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]
approve

**Pull request merged**